src/ciphersuite/mod.rs

```diff
@@ -195,9 +195,9 @@ impl Ciphersuite {
         hash(self.hash, payload)
     }
 
-    /// Get the length of the used hash algorithm.
-    pub(crate) fn hash_length(&self) -> usize {
-        get_digest_size(self.hash)
+    /// Get the output length of the kdf.
+    pub(crate) fn hkdf_length(&self) -> usize {
+        get_tag_size(self.hmac)
```

Comment:
The MLS spec mentions both `KDF.Nsk` and `KDF.Nh`. Are the two the same?

Reply:
Where does the MLS spec mention `KDF.Nsk`? I couldn't find it. There is an `Nsk` in the HPKE RFC. Is that what you meant? There is also `AEAD.Nk` and `AEAD.Nn`.

Reply:
It is mentioned [here](https://github.com/mlswg/mls-protocol/pull/338/files), but it seems that it got reverted in a later PR. I think it's in #71.

There also seems to be some overlap with #57.

It looks to me like it would be good to combine #46, #57 and #71.

Reply:
We could do that, but I don't think it hurts to just merge this PR and then tackle the other issues based on the changes here.

Reply:
It's just that this PR is addressing the spec changes that introduce `KDF.Nsk`, but that's not what's implemented.

Reply:
I see. How should we proceed then? Should I use this PR to also tackle the other issues? Alternatively, I create a meta issue that gathers the changes needed for #46, #57 and #71. Then this PR solves one item on that meta-issue todo list.

Reply:
How you want to do it is up to you. Personally I'd extend this PR to cover the two other issues, unless it gets too big.

Reply:
It's up to me, but I'm also trying to learn what "good practice" would look like or what good options are to organize work in this style of workflow. But ok, I'll continue work on this PR then and will split it once it gets too big.